pymatgen/io/lammps/inputs.py

```diff
@@ -46,58 +150,54 @@ def get_string(self):
 
         """
 
-        lammps_input = '## LAMMPS input generated from LammpsInputConstructor\n\n'
+        lammps_input = "## LAMMPS input generated from LammpsInputConstructor\n\n"
         for stage, command_dict in self.input_settings.items():
             for command, args in command_dict.items():
-                if command.startswith('comment') or command.startswith('header'):
-                    if command.startswith('header'):
-                        lammps_input += '\n'
-                    lammps_input += self.add_string(args)
+                if command.startswith("comment") or command.startswith("header"):
+                    if command.startswith("header"):
+                        lammps_input += "\n"
+                    lammps_input += args + "\n"
                 else:
-                    lammps_input += self.format_command(command, args)
+                    lammps_input += f"{command:20} {args}\n"
         return lammps_input
 
-    @staticmethod
-    def format_command(command, args):
+    def from_string(self, s: str):  # type: ignore
         """
-        Simple format helper for adding command with the arguments.
+        Helper method to parse string representation of LammpsInputFile
 
         Args:
-            command: LAMMPS command.
-            args: Arguments for LAMMPS command.
-
-        Returns:
+            s: String representation of LammpsInputFile.
 
+        Returns: LammpsInputFile
         """
 
-        return "{0:20} {1}\n".format(command, args)
+        for line in self._clean_lines(s.splitlines()[1:]):
+            if line[:2] == "##":
+                self.add_comment(comment=line[2:], is_stage_header=False)
+            if line[0] == "#":
+                self.init_stage()
+                self.add_comment(comment=line[2:], is_stage_header=True)
+            else:
+                self.add_commands(line)
 
-    @staticmethod
-    def add_string(string):
-        """
-        Helper method for adding string to lammps input file.
-        Args:
-            string: String to be added to lammps input file.
-        """
-        # return '\n' + string + '\n'
-        return string + '\n'
+        return LammpsInputFile(self.input_settings)
 
-    @staticmethod
-    def from_string(s):
+    def from_file(self, path: Union[str, Path]):  # type: ignore
         """
-        Reads LammpsInputFile from string.
+        Creates an InputFile object from a file.
 
         Args:
-            s: LammpsInputFile string.
+            path: Filename to read, including path.
 
         Returns:
-            LammpsInputFile object
+            InputFile
         """
-        constr = LammpsInputConstructor()
-        return constr._from_string(s)
+        filename = path if isinstance(path, Path) else Path(path)
+        with zopen(filename, "rt") as f:
+            return self.from_string(f.read())
```

Comment:
Again apologies if we've discussed this already, but what is it that's preventing you from using the inherited `from_file` method here? I don't see anything here that should be specific to the internal structure (i.e., `OrderedDict` of `LammpsInputFile`.